lib/middleware/websocket/controller.rb

```diff
@@ -39,18 +42,35 @@ def on_close(connection)
     private
 
     def generate_client(connection:)
-      @clients << Client.new(connection: connection)
-      @clients.last.generate_player
+      room = find_room(connection: connection)
+      client = Client.new(connection: connection, room_id: room.id)
+      room.add_client(client: client)
+      client.generate_player
+      @clients << client
 
       Interactor::ClientCreationUpdate.new.call(client: @clients.last)
     end
 
     def find_client(connection:)
-      @clients.select { |client| client.connection == connection }.first
+      @clients.detect { |client| client.connection == connection }
     end
 
     def race_update
       @race_update ||= Interactor::RaceUpdate.new
     end
+
+    def find_room(connection:)
```

Comment:
⚒ Generally a method name `find` won't create anything. We should name this `find_or_create_room` instead, as the current name is misleading.

Reply:
@jgayfer I'm actually just re-visiting this and I've re-named it `setup_room`, maybe a bit better